### PR TITLE
Allow groups as change request approvers

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [X] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [X] I have filled in the "Changes" section below?
- [X] I have filled in the "How did you test this code" section below?

## Changes

Now users can assign groups as reviewers/approvers for change requests.

## How did you test this code?
Manually,

Create an environment and a feature, create some users and groups.

- Configure the enviroment to use change requests.
- change the feature and assign some users and groups as reviewers.
- Edit the change request.

For this feature a couple of flags have been created:

- `enable_groups_as_reviewers` : If this feature flag is set to `true`, this flag will show the Assigned groups section in the Change Requests and in the Create Change Request modal in the Features page.
- `disable_users_as_reviewers`: This flag has been named and used in its negative form for backward compatibility. If it is set to `false` and `enable_groups_as_reviewers` is also false, the app will behave as before these changes. If the flag is set to true,  the Assigned users section in the Change Requests and in the Create Change Request modal in the Features page will be hidden.

### CODE REVIEW

File: frontend/web/components/groupSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  const [filter, setFilter] = useState<string>('')
+  const grouplist =
+    groups &&
+    groups.filter((v) => {
+      const search = filter.toLowerCase()
+      if (!search) return true
+      return `${v.name}`.toLowerCase().includes(search)
+    })
+  return (
+    <InlineModal
```
Comment (kyle-ssg) [inline frontend/web/components/groupSelect.tsx]: Upper case filename GroupSelect.tsx

**Comment by matthewelwell** [comment]
Seems like the feature flags `groups_as_reviewers` is not working as expected - can still see groups option when flag is enabled. Seems like it just prevents the 'Add groups' link being clicked, vs hiding the option entirely.

File: frontend/web/components/groupSelect.tsx (hunk 1 above)
Comment (novakzaballa) [reply frontend/web/components/groupSelect.tsx]: Done

**Comment by matthewelwell** [comment]
image

Slight UI issue here - don't need to add `Groups: ` before the list of assigned groups.

**Comment by dabeeeenster** [comment]
I cant get these cogs to work? 
image

**Comment by novakzaballa** [comment]
> I cant get these cogs to work?
> image

@dabeeeenster these cogs have been replaced by the `Add user` and `Add group` buttons to provide a more consistent UX with the Create Change Request modal.

**Review by matthewelwell — DISMISSED** [review]
This looks good to me now!

**Comment by matthewelwell** [comment]
@novakzaballa as mentioned in Slack, adding the `users_as_reviewers` flag will mean that all existing self hosted customers will have to enable the flag to maintain the current functionality. This will be a bit of a nightmare to manage. I appreciate there was reasoning behind adding that flag but I think that after all, we should not add the flag and leave the users as reviewers as functionality that is always available.

An alternative would be to change the flag name to `disable_users_as_reviewers` and invert all of the conditional checks. This is probably easier?

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**